src/amrex/ParticleContainer.py

```diff
@@ -0,0 +1,104 @@
+"""
+This file is part of pyAMReX
+
+Copyright 2023 AMReX community
+Authors: Axel Huebl
+License: BSD-3-Clause-LBNL
+"""
+
+
+def pc_to_df(self, local=True, comm=None, root_rank=0):
+    """
+    Copy all particles into a pandas.DataFrame
+
+    Parameters
+    ----------
+    self : amrex.ParticleContainer_*
+        A ParticleContainer class in pyAMReX
+    local : bool
+        MPI-local particles
+    comm : MPI Communicator
+        if local is False, this defaults to mpi4py.MPI.COMM_WORLD
+    root_rank : MPI root rank to gather to
+        if local is False, this defaults to 0
+
+    Returns
+    -------
+    A concatenated pandas.DataFrame with particles from all levels.
+
+    Returns None if no particles were found.
+    If local=False, then all ranks but the root_rank will return None.
+    """
+    import pandas as pd
+
+    # create a DataFrame per particle box and append it to the list of
+    # local DataFrame(s)
+    dfs_local = []
+    for lvl in range(self.finest_level + 1):
+        for pti in self.const_iterator(self, level=lvl):
+            if pti.size == 0:
+                continue
+
+            if self.is_soa_particle:
+                next_df = pd.DataFrame()
+            else:
+                # AoS
+                aos_np = pti.aos().to_numpy(copy=True)
+                next_df = pd.DataFrame(aos_np)
+                next_df.set_index("cpuid")
+                next_df.index.name = "cpuid"
+
+            # SoA
+            soa_view = pti.soa().to_numpy(copy=True)
+            soa_np_real = soa_view.real
+            soa_np_int = soa_view.int
+
+            for idx, array in enumerate(soa_np_real):
+                next_df[f"SoA_real_{idx}"] = array
+            for idx, array in enumerate(soa_np_int):
+                next_df[f"SoA_int_{idx}"] = array
+
```

Comment:
todo: I think this might miss on the runtime attributes (also are SoAs).

This should go into `SoA::to_numpy` aka `soa_to_numpy` & `soa_to_cupy`,

Reply:
Requires to call:
- `AddRealComp`
- query current number of real components
- call `ResizeRuntimeRealComp` again with redundant info (again the comm and number of components)

but https://github.com/AMReX-Codes/amrex/pull/3615  could simplify this.